Articles/WhatIs/WhatIsPeerCodeReview.md

```diff
@@ -0,0 +1,61 @@
+
+<!--- deck end --->
+
+<!--- body start --->
+
+Peer code reviews are an essential part of the software development process where, along with software testing, provide the foundation for software quality assurance.
+
+There are many benefits to performing peer code reviews, including:
+
+* Improved code quality: Peer code reviews can help to find and fix defects in code that would otherwise go undetected.
+This can lead to improved software quality and reduced risk of defects.
+
+* Increased security: Peer code reviews can help to identify and fix security vulnerabilities in code.
+This can help to protect software from attack.
+
+* Improved performance: Peer code reviews can help to identify and fix performance bottlenecks in code.
+This can improve the performance of software and make it more responsive to users.
+
+* Reduced development costs: Peer code reviews can help to reduce the cost of software development by finding and fixing defects early in the development process and improve the maintainability and sustainability of the code.
+This can save time and money that would otherwise be spent on debugging and fixing defects later in the development process.
+
+* Improved developer skills: Peer code reviews can help developers to learn from each other and improve their coding skills.
+This can lead to improved productivity and quality of code.
+
+* Improved team collaboration: Peer code reviews can help to improve team collaboration and communication.
+This can lead to a more cohesive and productive team.
+
+There are a few things to keep in mind when conducting a peer code review:
+
+* **Set clear expectations.** Before the review begins, the author of the code should clearly explain the purpose of the code and the specific areas that they would like the reviewer to focus.
+
+* **Be specific.** When providing feedback, be as specific as possible. This will help the author of the code to understand the issue and how to fix it.
+
+* **Be respectful.** Remember that the author of the code is putting their work out there for review.
+Be respectful of their time and effort, and avoid making personal attacks.
+
+* **Be constructive.** The goal of a peer code review is to improve the code, not to tear it down.
+Focus on providing constructive feedback that will help the author of the code to improve their code.
+
+* **Be open to feedback.** The author of the code should be open to feedback from the reviewer.
+Even if they don't agree with all of the feedback, they should be willing to consider it and make changes to their code as needed.
+
+In summary, peer code reviews are an essential part of the software development process.
+They can help to improve the quality of code, reduce the risk of defects, and improve the overall security of software.
+If you're not already conducting peer code reviews, I encourage you to start. You'll be glad you did!
+
+<!--- body end  --->
+ 
+<!---
+Publish: yes
+Pinned: no
+Topics: peer code review
```

Comment:
So, this is interesting. I was looking for this topic and expected to find it in **Better Collaboration**. I don't. Instead I find it in **Better Reliability** with this definition...

> **Peer Code Review:** The collaborative process of ensuring software meets quality, readability, and maintainability requirements.

And its definition begins with the word *collaborative*.

Reply:
Right, but because code reviews are such an important foundation of software quality assurance one can argue that if you had to pick between **Better Reliability** and **Better Collaboration**, one would pick  **Better Reliability**.  The collaborative part is a byproduct, not the primary goal.  In Steve McConnell's book "Code Complete: 2nd Edition", the says:

> The primary purpose of collaborative construction is to improve software quality.

where by "collaborative construction", he means peer code review (of which pair-programming is one form of collaborative construction and means the criteria for peer code review).

P.S. That is a great book by the way if you have never read it (see my summary [here](https://bartlettroscoe.github.io/reading-list/#code_complete_2nd_edition_2004)).